File: FindTheDifference/FindTheDifference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    var map = new Map();
    for (var i = 0; i < s.length; i++) {
        var b = s.charAt(i);
        if(!map.has(b))
            map.set(b, 1);
        else map.set(b, map.get(b) + 1);
    }
    for (var i = 0; i < t.length; i++) {
        var c = t.charAt(i);
        if(map.has(c) && map.get(c) > 0)
            map.set(c, map.get(c) - 1);
        else return c;
    }
};

File: src/FindTheDifference/FindTheDifference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int charCode = t.charAt(s.length());
        for(int i = 0; i < s.length(); i++) {
            charCode += (int)t.charAt(i);
            charCode -= (int)s.charAt(i);
        }
        return (char)charCode;
    }
}